### ISSUE TITLE
Emit Mutation Specific QVs

### ISSUE THREAD

**Issue by evolvedmicrobe** [PacificBiosciences/pbccs#35]
Right now our QV values are an underdetermined representation of the possible mutations.  That is, given a QV value one cannot tell how the error probability is distributed amongst the deletion, insertion and substitution values.

This creates problems both for using downstream applications, and for figuring out how to reverse complement reads.  For example, consider the following alignment in the forward orientation.

```
99099999
ATGGGGAT
AT-GGGAT
```

Here the first base in the HP has a low QV, presumably due to the deletion.  Now the reverse complemented left aligned indel looks like this:

```
99999099
ATCCCCTA
AT-CCCTA    
```

And now we have a very high QV value associated with the deletion.  One solution would be to simply swap the QV Value associated with the first and last bp in the homopolymer.  This generally works, however there are edge cases like the following which have appeared.

```
9909
ATTA
ATCA 
```

Where the SNP has the correct QV in the forward orientation, but after applying a "reverse and swap homopolymers" rule, looks like this:

```
9909
TAAT
TGAT
```

Leading us to call the SNP with our highest confidence.  Now, we can apply yet another heuristic to solve this case (only swap if the first QV is less than the last in the forward orientation), however, I think this is slowing moving us in the direction of a nasty mess of heuristics, rules, etc. that our customers are almost surely not to know.

I think we need to sit down and figure out how we emit QVs and how we expect them to be consumed.  Some ideas off the top of my head:
-  Get an aligner to handle reverse complementing appropriately
-  Emit rich QVs for insertion/deletion substitutions
-  Emit rich QVs that conform to some other framework (GATK, etc.) so we can ease consumption.

**Comment by dalexander** [comment]
I concur with Lance that no one should be attempting to make sense of our QVs in genomic orientation---they only make sense in the native orientation.  The tools that do this right:  just Quiver.

Is your argument is that you would like the QVs more accessible to "generic" analysis tools that do pileup-type stuff, I agree we could perhaps do a better job there... we could come up with some heuristic means of making the QVs more meaningful in genomic orientation.... but this might be a losing battle

**Comment by dalexander** [comment]
You didn't mention the option of distributing the uncertainty along the span of the homopolymer.

**Comment by evolvedmicrobe** [comment]
So, the history of this is that ccscheck was making QV calibration plots by doing this very naive type of analysis where you align the CCS read, and then get the QV value that corresponds to any error that you find.  I noticed that even at our highest accuracy very rare errors still occurred.  Closer inspection showed it was always due to some edge case in how QV values are assigned like what is described here.  It looked to me that for these cases, if a variant caller that specifically compared the odds of one template to another was used instead of just a single marginal QV value, weird stuff like this wouldn't happen.

ccscheck is a generic pile-up caller, and as I thought this would apply to other generic pile-up callers I opened this issue (I don't think we have a good variant calling solution right now so wanted to support that type of thing).

However, though doing these type of heuristics cleans up a lot of the problems in the pile-up approach, I think in order to get to QV Infinity at the highest yield one will need to specifically compare template likelihoods rather than per-base QV values.  I am not sure thinking too much about this makes sense without an application in mind though.  As you said, it's a bit of a losing battle.
